tools/codegen/config.yml

```diff
@@ -336,75 +192,147 @@ resources:
+    version_header: application/vnd.atlas.2023-01-01+json
 
-  # stream_privatelink_endpoint:
-  #   read:
-  #     path: /api/atlas/v2/groups/{groupId}/streams/privateLinkConnections/{connectionId}
-  #     method: GET
-  #   create:
-  #     path: /api/atlas/v2/groups/{groupId}/streams/privateLinkConnections
-  #     method: POST
-  #   schema:
-  #     aliases:
-  #       group_id: project_id
-  #       _id: id
-  #     ignores: ["links"]
+  resource_policy_api:
+    read:
+      path: /api/atlas/v2/orgs/{orgId}/resourcePolicies/{resourcePolicyId}
+      method: GET
+    create:
+      path: /api/atlas/v2/orgs/{orgId}/resourcePolicies
+      method: POST
+    update:
+      path: /api/atlas/v2/orgs/{orgId}/resourcePolicies/{resourcePolicyId}
+      method: PATCH
+    delete:
+      path: /api/atlas/v2/orgs/{orgId}/resourcePolicies/{resourcePolicyId}
+      method: DELETE
+    schema:
+      aliases:
+        resource_policy_id: id # path param name does not match the API response property
 
-  # flex_snapshot:
-  #   # does not have a POST endpoint so schema can't be autogenerated
-  #   create:
-  #     path: /api/atlas/v2/*
-  #     method: POST
-  #   schema:
-  #     aliases:
-  #       group_id: project_id
-  #     ignores: ["links"]
+  search_deployment_api:
+    read:
+      path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}/search/deployment
+      method: GET
+    create:
+      path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}/search/deployment
+      method: POST
+      wait: &search-deployment-create-wait
+        state_property: "stateName"
+        pending_states: ["UPDATING", "PAUSED"]
+        target_states: ["IDLE"]
+        timeout_seconds: 10800 # 3 hours
+        min_timeout_seconds: 60
+        delay_seconds: 60
+    update:
+      path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}/search/deployment
+      method: PATCH
+      wait: *search-deployment-create-wait
+    delete:
+      path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}/search/deployment
+      method: DELETE
+      wait:
+        <<: *search-deployment-create-wait
+        min_timeout_seconds: 30
+        pending_states: ["IDLE", "UPDATING", "PAUSED"]
+        target_states: ["DELETED"] # DELETED is a special state value when API returns 404 or empty object
+    version_header: application/vnd.atlas.2025-03-12+json
 
-  # flex_restore_job:
-  #   read:
-  #     path: /api/atlas/v2/groups/{groupId}/flexClusters/{name}/backup/restoreJobs/{restoreJobId}
-  #     method: GET
-  #   create:
-  #     path: /api/atlas/v2/groups/{groupId}/flexClusters/{name}/backup/restoreJobs
-  #     method: POST
-  #   schema:
-  #     aliases:
-  #       group_id: project_id
-  #     ignores: ["links"]
+  # Update doesn't work because OpenAPI spec defines incorrectly region and cloud_provider at root level in PATCH instead of inside data_process_region as in the other endpoints.
```

Comment:
```suggestion
  # Update doesn't work because API defines incorrectly region and cloud_provider at root level in PATCH instead of inside data_process_region as in the other endpoints.
```